assets/js/modules/analytics/util/report-validation.js

```diff
@@ -57,3 +57,38 @@ export function isValidDimensions( dimensions ) {
 		return dimension.hasOwnProperty( 'name' ) && typeof dimension.name === 'string';
 	} );
 }
+
+/**
+ * Verifies provided dimensionFilters to make sure they match allowed values found in dimensions.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}                               dimensionFilters The dimension filters to check.
+ * @param {string|string[]|Object|Object[]|null} dimensions       The dimensions to check.
+ * @return {boolean} TRUE if dimension filters are valid, otherwise FALSE.
+ */
+export function isValidDimensionFilters( dimensionFilters, dimensions ) {
+	if ( ! dimensions ) {
+		return false;
+	}
+	const validDimensions = [];
+
+	// Grab the dimension from a single dimension string/object.
+	const parseDimension = ( dimension, dimensionsCollection ) => {
+		if ( typeof dimension === 'string' ) {
+			dimensionsCollection.push( dimension );
+		} else if ( typeof dimension === 'object' && dimension.hasOwnProperty( 'name' ) ) {
+			dimensionsCollection.push( dimension.name );
+		}
+	};
+
+	if ( Array.isArray( dimensions ) ) {
+		dimensions.forEach( ( singleDimension ) => parseDimension( singleDimension, validDimensions ) );
+	} else {
+		parseDimension( dimensions, validDimensions );
+	}
+
+	// Ensure every dimensionFilter key corresponds to a valid dimension and ensure every value is a string.
+	return Object.keys( dimensionFilters ).every( ( dimensionFilterKey ) => validDimensions.include( dimensionFilterKey ) ) &&
```

Comment:
I believe we can simplify it to this:

```suggestion
	return Object.keys( dimensionFilters ).every( validDimensions.include, validDimensions ) &&
```